driver-core/src/test/java/com/datastax/driver/core/TableMetadataTest.java

```diff
@@ -474,4 +482,53 @@ public void should_not_mix_indexes_from_different_tables() {
         assertThat(table_cd.getIndexes().size()).isEqualTo(1);
         assertThat(table_cd.getIndexes().get(0).getName()).isEqualTo("test_d");
     }
+
+    /**
+     * Validates that the 'extensions' option is properly parsed when set.  This value is currently not
+     * modifiable via CQL so we fake out a row containing populated extensions by retrieving
+     * the current metadata and changing the extensions column.  The row is passed to the Options
+     * constructor and we validate the extensions map is matches what was set.
```

Comment:
Another possible approach is to mutate the row in the system table with a CQL query:
```
update system_schema.tables set extensions = {'foo': 0x00}
where keyspace_name = 'test' and table_name = 'foo';
```
I'm not sure if it's simpler because we then need to trigger a reload (I don't think the above query triggers a notification). We could add a column to the table.